### PR TITLE
Break `ObjectPattern`s with nested `ObjectPattern`s

### PR DESCRIPTION
**Pull request by duailibe**
Context: #2550

After significant demand, this PR introduces an heuristic to always break `ObjectPattern`s if there are any nested `ObjectPattern` (or `ArrayPattern` in that case) properties.

I chose not to use the "always break if any property has a value" for now and not to break the outermost `ObjectPattern` if it's a function argument.

I personally think it's a good compromise, but I didn't mark to close the issue yet.

Let me know what you think!

### CODE REVIEW

File: tests/destructuring/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -4,9 +4,56 @@ exports[`destructuring.js 1`] = `
...
+
+const { a, b, c, d: { e } } = someObject;
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 const [one, two = null, three = null] = arr;
 a = ([s = 1]) => 1;
 const { children, ...props } = this.props;
 
+const {
+  user: { firstName, lastName }
+} = this.props;
```
Comment (duailibe) [inline tests/destructuring/__snapshots__/jsfmt.spec.js.snap]: I'm on the fence with this one... I personally prefer it collapsed, but don't care either way

**Comment by duailibe** [comment]
@tennisgent Would you open a new issue with this please?

**Comment by tennisgent** [comment]
Certainly. Thanks for the quick response. 👍

**Comment by amandakoster** [comment]
Hi Everyone,  I am using Eslint and Prettier, and am wondering if there is a Prettier rule I can add to my .eslintrc file to keep the destructure multi line vs. single line? Thank you!

**Comment by j-f1** [comment]
@amandakoster Unfortunately, we don’t offer those kinds of options. Sorry :/

**Comment by tennisgent** [comment]
As per https://github.com/prettier/prettier/issues/4384, the maintainers decided not to make an option for this. So it's all or nothing. Not configurable.

**Pull request merged**